dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBVectorStoreRecordMapper.cs

```diff
@@ -68,84 +66,18 @@ public AzureCosmosDBMongoDBVectorStoreRecordMapper(VectorStoreRecordDefinition v
         // Use Mongo reserved key property name as storage key property name
         this._storagePropertyNames[keyProperty.Name] = AzureCosmosDBMongoDBConstants.MongoReservedKeyPropertyName;
 
-        this._reversedStoragePropertyNames = ReverseDictionary(storagePropertyNames);
-    }
-
-    public BsonDocument MapFromDataToStorageModel(TRecord dataModel)
-        => MapDocument(dataModel.ToBsonDocument(), this._storagePropertyNames);
-
-    public TRecord MapFromStorageToDataModel(BsonDocument storageModel, StorageToDataModelMapperOptions options)
-        => BsonSerializer.Deserialize<TRecord>(MapDocument(storageModel, this._reversedStoragePropertyNames));
-
-    #region private
-
-    private static BsonDocument MapDocument(BsonDocument document, Dictionary<string, string> propertyMappings)
-    {
-        var newDocument = new BsonDocument();
-
-        foreach (var element in document)
-        {
-            BsonValue newValue;
-            if (element.Value.IsBsonDocument)
-            {
-                newValue = MapDocument(element.Value.AsBsonDocument, propertyMappings);
-            }
-            else if (element.Value.IsBsonArray)
-            {
-                newValue = MapDocumentInArray(element.Value.AsBsonArray, propertyMappings);
-            }
-            else
-            {
-                newValue = element.Value;
-            }
-
-            if (propertyMappings.TryGetValue(element.Name, out var newName))
-            {
-                newDocument[newName] = newValue;
-            }
-            else
-            {
-                newDocument[element.Name] = newValue;
-            }
-        }
-
-        return newDocument;
-    }
-
-    private static BsonArray MapDocumentInArray(BsonArray array, Dictionary<string, string> propertyMappings)
-    {
-        var newArray = new BsonArray();
-
-        foreach (var item in array)
+        var conventionPack = new ConventionPack
         {
-            if (item.IsBsonDocument)
-            {
-                newArray.Add(MapDocument(item.AsBsonDocument, propertyMappings));
-            }
-            else if (item.IsBsonArray)
-            {
-                newArray.Add(MapDocumentInArray(item.AsBsonArray, propertyMappings));
-            }
-            else
-            {
-                newArray.Add(item);
-            }
-        }
+            new IgnoreExtraElementsConvention(ignoreExtraElements: true),
+            new AzureCosmosDBMongoDBNamingConvention(this._storagePropertyNames)
+        };
 
-        return newArray;
+        ConventionRegistry.Register(nameof(AzureCosmosDBMongoDBVectorStoreRecordMapper<TRecord>), conventionPack, type => true);
```

Comment:
Considering that this seems to be a global registration, do we run the risk of clashes, e.g. if someone uses two classes with a property name that happens to be the same but different bson names in each class.
Name => "hotel_name" in a Hotel class
Name => "name" in a CarModel class

It looks like the third parameter there might be a filter.  Can we use that to check if the type is equal to typeof(TRecord)?

Reply:
Good catch, added. I wasn't sure what is the purpose of that filter, but it makes sense now:
![image](https://github.com/user-attachments/assets/6cdd17b2-64e6-4b75-a9ad-e184e9b5685e)